.github/workflows/set-date.yaml

```diff
@@ -0,0 +1,29 @@
+name: Set Date
+
+on:
+  schedule:
+    - cron: '* 0 * * *'
+
+jobs:
+  update-date:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+
+    - name: Write Date File
+      run: |
+        echo -e -n "#ifndef VCS_DATE\n#define VCS_DATE $(date "+%Y%m%d")\n#endif" > ./firmware/date_stamp.h
+        echo -e -n "#ifndef VCS_DATE\n#define VCS_DATE $(date "+%Y%m%d")\n#endif" > ./simulator/date_stamp.h
```

Comment:
why three separate files/dates? let's have one file for all consumers?

Reply:
It was easier.

Reply:
If you'd prefer to have one file, your wish is my command.

Reply:
Let's have one single file next to engine_controller.h

Reply:
In the same directory?

Reply:
Maybe the util directory would be better?

Reply:
Any directory zero difference to me, as long as one file and it compiles :)